plugins/woocommerce/includes/class-wc-coupon.php

```diff
@@ -1152,4 +1166,26 @@ public function set_short_info( string $info ) {
 		$this->set_amount( $info[3] ?? 0 );
 		$this->set_free_shipping( $info[4] ?? false );
 	}
+
+	/**
+	 * Returns alternate error messages based on context (eg. Cart and Checkout).
+	 *
+	 * @param int $err_code Message/error code.
+	 *
+	 * @return array Context based alternate error messages.
+	 */
+	public function get_context_based_coupon_errors( $err_code = null ) {
+
+		switch ( $err_code ) {
+			case self::E_WC_COUPON_NOT_YOURS_REMOVED:
+				return array(
+					/* translators: %s: coupon code */
+					'cart'     => sprintf( __( 'Please enter a valid email at checkout to use coupon code "%s".', 'woocommerce' ), esc_html( $this->get_code() ) ),
+					/* translators: %s: coupon code */
+					'checkout' => sprintf( __( 'Please enter a valid email to use coupon code "%s".', 'woocommerce' ), esc_html( $this->get_code() ) ),
+				);
```

Comment:
On my end, I'm seeing the cart error message also on the checkout page, when using the checkout shortcode:

<img width="1096" alt="Screenshot 2024-03-13 at 16 20 09" src="https://github.com/woocommerce/woocommerce/assets/3323310/ac102bae-a67d-4bc6-ad26-d891265ec23d">

This is independent of weather I define the checkout shortcode page as the default checkout page or not. The issue becomes visible when I do not have an email address entered or when the entered email address does not match the email address of the coupon.

> [!NOTE]
> I've named the email coupon `foo` and used the email address foo@bar.com, as I can type that faster. So please don;t wonder why you're not seeing the coupon code `exclusive` in my screenshot.

Reply:
Can you please run `pnpm clean` and then install && build again?